### PR TITLE
Arr/join new circle

### PR DESCRIPTION
**Pull request by Statmosphere**
**Changes**
Includes a new page for users to scan or search for a new circle.

**Notes**
I couldn't get the expo-camera QR scanner to work so I used another expo library.

**YouTrack Link**
https://prayercircle.youtrack.cloud/issue/PC-132/Add-Circle-Page

### CODE REVIEW

File: app/filter.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
-import React from 'react';
-import { SafeAreaView, Text, View, StatusBar } from 'react-native';
-import { styled } from 'nativewind';
-import { Button } from '../components/Buttons';
+import React from "react";
```
Comment (dev-arekusandoru) [inline app/filter.js:1]: change your formatter in settings to use single quotes, otherwise there will be a lot of these quote replacements happening. Our standard for development is single quotes

File: app/joinCircle.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    if (hasPermission === null) {
+        return <Text>Requesting for camera permission</Text>;
+    }
+    if (hasPermission === false) {
+        return <Text>No access to camera</Text>;
+    }
+
+    return (
+        <StyledSafeArea className="bg-offblack border" style={{ flex: 1 }}>
```
Comment (dev-arekusandoru) [inline app/joinCircle.js]: `border` here is not needed

Comment (dev-arekusandoru) [reply app/joinCircle.js]: you can also remove the style tag and add `flex-1` into className

File: app/joinCircle.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            <KeyboardAwareScrollView>
```
Comment (dev-arekusandoru) [inline app/joinCircle.js]: add `bounces={false}` here to remove make the page less jumpy with the scrolling

File: package.json [hunk 4]
```diff
@@ -13,8 +13,10 @@
     "@expo/react-native-action-sheet": "^4.0.1",
     "@expo/webpack-config": "^19.0.0",
     "@react-native-async-storage/async-storage": "1.18.2",
+    "@shopify/flash-list": "1.4.3",
```
Comment (dev-arekusandoru) [inline package.json]: run `npm remove @shopify/flash-list` it is no longer needed in the project and has been causing errors on `main`

File: backend/config.js [hunk 5]
```diff
@@ -1,34 +1,34 @@
 // Import the functions you need from the SDKs you need
-import { initializeApp } from 'firebase/app';
-import { getDatabase } from 'firebase/database';
+import { initializeApp } from "firebase/app";
```
Comment (dev-arekusandoru) [inline backend/config.js]: None of the changes here should be new. Run `git merge main` from your branch to get all the new changes. This will help with merge conflicts as well. Please let me know if there are conflicts that you are unsure of, rather than just guessing

File: app/register.js [hunk 6]
```diff
@@ -1,24 +1,24 @@
-import React, { useState, useRef } from 'react';
+import React, { useState, useRef } from "react";
```
Comment (dev-arekusandoru) [inline app/register.js]: not sure what changes you would be making on this page, please review and remove any changes that are not related to this Item. also, by fixing your formatter, a lot of changes will go away related to quotations

File: app/joinCircle.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                        className="bg-offblack text-[18px] flex-1 w-[300px] h-[42px] text-offwhite border border-offwhite rounded-lg px-3 py-[5px] mr-1"
+                        placeholder={"Code:"}
+                        placeholderTextColor={"#ffffff66"}
+                        inputMode="numeric"
+                        maxLength={8}
+                        onChangeText={(text) => {
+                            setCode(text);
+                        }}
+                        ref={(input) => {
+                            this.searchCode = input;
```
Comment (dev-arekusandoru) [inline app/joinCircle.js]: it doesn't look like you have set this variable anywhere. To do so, add `const searchCode = useRef();` in the code above the return statement

**Review by dev-arekusandoru — CHANGES_REQUESTED** [review]
I just looked over the code and left some comments on things I noticed. I wasn't able to actually test it out so when I do I'll leave another comment. Please make the changes listed in the code and re-request review afterwards. If you have any questions, don't hesitate to reach out!

**Comment by Statmosphere** [comment]
I have made the changes you have requested. How does it run on your end?

**Review by dev-arekusandoru — APPROVED** [review]
Looks good! Go ahead and merge into master

**Comment by dev-arekusandoru** [comment]
@Statmosphere one last formatter thing i noticed. make sure your tab size is set to 4

**Comment by aidan-hubley** [comment]
These views need revisited for other screen sizes.
